### PR TITLE
Run Mac x64 build tests in postsubmit only

### PR DESCRIPTION
**Pull request by jmagman**
It shouldn't be very common for x64 example projects to not build if they built on arm64.  We have other tool tests that would probably catch those issues in presubmit.

Instead run them on postsubmit only to free up limited x64 staging resources.

Follow up from #141206

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by jmagman** [comment]
@godofredoc I should have thought of this as part of https://github.com/flutter/flutter/pull/141206 🙂

File: .ci.yaml [hunk 1]
```diff
@@ -3180,6 +3180,7 @@ targets:
 
   - name: Mac_x64 build_tests_1_4
     recipe: flutter/flutter_drone
+    presubmit: false
```
Comment (christopherfujino) [inline .ci.yaml:3183]: should we leave a comment why? I could foresee in the future someone cargo-culting this change to the arm64 ones.

Comment (jmagman) [reply .ci.yaml]: Added a comment.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**